### PR TITLE
feat: add shorebird_code_push package

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds `package:shorebird_code_push`, which will expose bindings to allow Flutter apps to interact with the Updater.

Part of https://github.com/shorebirdtech/shorebird/issues/487

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

**Comment by bryanoltman** [comment]
@felangel lmk if you think I should remove any of the autogenerated files. I'm new to this whole very_good_dart_package thing :)

**Comment by eseidel** [comment]
Have you thought about the coupling?  What versions of the Rust code (or Shorebird server) will this work with?  Does it need to be tied closer to either of those by being next to them in the same repo?

**Comment by eseidel** [comment]
Oh, also, the code exists here: https://github.com/shorebirdtech/updater/tree/main/dart_bindings. Feel free to move it (which it looks like you're doing), just I'd recommend documenting your intended solutions for coupling as part of doing so.

**Comment by felangel** [comment]
> @felangel lmk if you think I should remove any of the autogenerated files. I'm new to this whole very_good_dart_package thing :)

Yup, I would delete the .github directory since this is part of a mono-repo with an existing .github directory.

File: packages/shorebird_code_push/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: shorebird_code_push
+description: Get info about your Shorebird code push app
+version: 0.1.0+1
+
```
Comment (felangel) [inline packages/shorebird_code_push/pubspec.yaml:4]: nit remove extra newline

File: packages/shorebird_code_push/.github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (felangel) [inline packages/shorebird_code_push/.github/ISSUE_TEMPLATE/bug_report.md]: The whole .github directory can be removed. Also we should update the existing .github workflow to run checks against this new package.

**Review by felangel — CHANGES_REQUESTED** [review]
See comments

File: packages/shorebird_code_push/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Shorebird Code Push
+
+[![License: MIT][license_badge]][license_link]
```
Comment (felangel) [inline packages/shorebird_code_push/README.md:3]: Looks like the links are missing here (license_badge and license_link)

File: packages/shorebird_code_push/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# 0.1.0+1
```
Comment (felangel) [inline packages/shorebird_code_push/CHANGELOG.md]: We can probably omit the CHANGELOG for now 🤷

File: packages/shorebird_code_push/pubspec.yaml (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_code_push/pubspec.yaml:4]: Done.

File: packages/shorebird_code_push/.github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_code_push/.github/ISSUE_TEMPLATE/bug_report.md]: Done

File: packages/shorebird_code_push/CHANGELOG.md (hunk 4 above)
Comment (bryanoltman) [reply packages/shorebird_code_push/CHANGELOG.md]: Done.

File: packages/shorebird_code_push/README.md (hunk 3 above)
Comment (bryanoltman) [reply packages/shorebird_code_push/README.md:3]: Fixed.

**Comment by bryanoltman** [comment]
Closing this in favor of https://github.com/shorebirdtech/updater/pull/30

**Pull request closed without merging**